hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -137,17 +160,69 @@ protected String tableName() {
 
   @Override
   public FileIO io() {
-    return fileIO;
+    if (encryptingFileIO != null) {
+      return encryptingFileIO;
+    }
+
+    encryptingFileIO = EncryptingFileIO.combine(fileIO, encryption());
```

Comment:
maybe no need to do this if `encryption()` is `PlaintextEncryptionManager.instance()` . Then we won't have to fix the tests with Hadoop config cast.